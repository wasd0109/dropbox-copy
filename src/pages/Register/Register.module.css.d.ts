declare const styles: {
  readonly "container": string;
  readonly "form": string;
  readonly "inputLabel": string;
  readonly "buttons": string;
  readonly "error": string;
  readonly "checkbox": string;
  readonly "checkboxLabel": string;
};
export = styles;

