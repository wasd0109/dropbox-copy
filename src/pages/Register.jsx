import React from "react";
import { Link } from "react-router-dom";

function Register() {
  return <Link to="/login">To login</Link>;
}

export default Register;
