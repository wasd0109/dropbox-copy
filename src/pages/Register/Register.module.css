.container {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}

.form {
  width: 80vw;
  padding: 4rem 4rem;
  --tw-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1),
    0 2px 4px -1px rgba(0, 0, 0, 0.06);
  box-shadow: var(--tw-ring-offset-shadow, 0 0 #0000),
    var(--tw-ring-shadow, 0 0 #0000), var(--tw-shadow);
}

.input {
  font-size: 2em !important;
}

.inputLabel {
  font-size: 4rem;
}

.buttons {
  margin-top: 2rem;
  display: flex;
  justify-content: space-between;
}

.buttons > button {
  font-size: 2.5rem;
  padding: 0.5rem 2rem;
}
.error {
  margin-top: 1rem;
}

.checkbox {
  margin-right: 1rem;
}

.checkbox > input {
  width: 2rem;
  height: 2rem;
}

.checkboxLabel {
  font-size: 2rem;
  margin: 0;
}
