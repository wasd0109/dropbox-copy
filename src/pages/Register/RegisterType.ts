export type RegisterFormData = {
  email: string;
  password: string;
  agreeTerm: boolean;
};
