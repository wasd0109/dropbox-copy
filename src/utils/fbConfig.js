const firebaseConfig = {
  apiKey: "AIzaSyCP4G9Y7gtFFloASOusc8qzi9ws_nzmMa0",

  authDomain: "dropbox-copy.firebaseapp.com",

  projectId: "dropbox-copy",

  storageBucket: "dropbox-copy.appspot.com",

  messagingSenderId: "431286804897",

  appId: "1:431286804897:web:c239d9981e4c8d5c7fabe0",

  measurementId: "G-WV8SBTT34J",
};

export default firebaseConfig;
