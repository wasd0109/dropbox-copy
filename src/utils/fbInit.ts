import firebase from "firebase";
import firebaseConfig from "./fbConfig";

firebase.initializeApp(firebaseConfig);

const auth = firebase.auth();
auth.useEmulator("http://localhost:9099");
const db = firebase.firestore();
db.useEmulator("localhost", 8080);

const storage = firebase.storage();
storage.useEmulator("localhost", 9199);

export { auth, db, storage };
