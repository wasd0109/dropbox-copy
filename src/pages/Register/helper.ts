import { RegisterFormData, RegisterFormError } from "./RegisterType";
import emailRegex from "../../validators/isEmail";

const findRegisterFormErrors = (form: RegisterFormData) => {
  const { email, password, agreeTerm } = form;
  const newErrors = {} as RegisterFormError;
  if (!email || email === "") newErrors.email = "Required";
  else if (!emailRegex.test(email))
    newErrors.email = "Please enter valid email";
  if (!password || password === "") newErrors.password = "Required";
  else if (password.length < 6)
    newErrors.password = "Password must be longer than 6 characters";
  if (!agreeTerm) newErrors.agreeTerm = "Please agree to Term of Services";
  return newErrors;
};

export default findRegisterFormErrors;
