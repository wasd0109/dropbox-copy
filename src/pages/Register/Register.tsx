import React, { useState } from "react";
import { useForm } from "react-hook-form";
import { RegisterFormData } from "./RegisterType";
import ErrorBar from "../../components/ErrorBar";
import ReactLoading from "react-loading";
import emailRegex from "../../components/validators/isEmail";
import { Link } from "react-router-dom";
import { auth } from "../../utils/fbInit";

function AuthForm() {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<RegisterFormData>();

  const [loading, setLoading] = useState(false);
  const [registerError, setRegisterError] = useState("");

  const onSubmit = async (data: RegisterFormData) => {
    setLoading(true);
    setRegisterError("");
    const { email, password } = data;
    try {
      await auth.createUserWithEmailAndPassword(email, password);
    } catch (err) {
      const authError = err as firebase.default.auth.Error;
      setLoading(false);
      setRegisterError(authError.message);
    }
  };

  return (
    <div className="grid h-screen auto-rows-auto grid-cols-1 gap-x-2">
      <form
        className="row-start-2 row-end-4 col-span-full flex flex-col shadow-md items-stretch py-4 px-4 mx-2 "
        onSubmit={handleSubmit(onSubmit)}
      >
        <label className="flex flex-col">
          Email
          <input
            type="email"
            className="border-2 mt-1"
            {...register("email", {
              required: "Required",
              pattern: {
                value: emailRegex,
                message: "Please enter a valid email",
              },
            })}
          />
        </label>
        {errors.email?.message && (
          <div className="mt-2">
            <ErrorBar message={errors.email.message} />
          </div>
        )}

        <label className="flex flex-col mt-4">
          Password
          <input
            type="password"
            className="border-2 mt-1"
            {...register("password", {
              required: "Required",
              minLength: {
                value: 6,
                message: "Password must be longer than 6 characters",
              },
            })}
          />
        </label>
        {errors.password?.message && (
          <div className="mt-2">
            <ErrorBar message={errors.password.message} />
          </div>
        )}

        <label className="flex mt-4">
          <input
            className="border-2 mt-1 mr-1"
            type="checkbox"
            {...register("agreeTerm", {
              required: "Please agree Term of Services",
            })}
          />
          <p>Agree to Term of Services</p>
        </label>
        {errors.agreeTerm?.message && (
          <div className="mt-2">
            <ErrorBar message={errors.agreeTerm.message} />
          </div>
        )}

        <div className="flex justify-center flex-col mt-4">
          {loading ? (
            <ReactLoading
              type="spin"
              color="#4299e1"
              className="self-center"
              width="2rem"
              height="2rem"
            />
          ) : (
            <input
              type="submit"
              aria-label="Register"
              value="Register"
              className="bg-blue-500 hover:bg-blue-700 cursor-pointer  text-white text-xl font-bold py-2 px-4 rounded self-center"
            />
          )}
          <Link
            to="/login"
            aria-label="To Login Page"
            className="self-end text-blue-500 hover:text-blue-700 cursor-pointer text-xl font-bold mt-2 mr-4"
          >
            Login
          </Link>
        </div>
        <div className=" self-center mt-2">
          {registerError ? <ErrorBar message={registerError} /> : null}
        </div>
      </form>
    </div>
  );
}

export default AuthForm;
