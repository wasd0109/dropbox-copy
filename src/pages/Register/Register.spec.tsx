import React from "react";
import { render, fireEvent } from "../../utils/testUtils";
import { RenderResult } from "@testing-library/react";
import Register from "./Register";
import { get } from "react-hook-form";

const TEST_EMAIL = "abc@gmail.com";
const TEST_PASSWORD = "123456";

let component: RenderResult;

beforeEach(() => {
  component = render(<Register />);
});

describe("Register page render correctly", () => {
  test("Register page render", () => {
    const { asFragment } = component;
    expect(asFragment).toMatchSnapshot();
  });

  test("Register page show Email input field", () => {
    const { getAllByLabelText } = component;
    expect(getAllByLabelText("Email").length).toEqual(1);
  });
  test("Register page show Password input field", () => {
    const { getAllByLabelText } = component;
    expect(getAllByLabelText("Password").length).toEqual(1);
  });

  test("Register page show Register button", () => {
    const { getAllByLabelText } = component;
    expect(getAllByLabelText("Register").length).toEqual(1);
  });

  test("Register page show Login link", () => {
    const { getAllByLabelText } = component;
    expect(getAllByLabelText("To Login Page").length).toEqual(1);
  });
});

describe("Register page function properly", () => {
  test("Email input allow input and change value", () => {
    const { getByLabelText, getAllByDisplayValue } = component;
    const emailInputRef = getByLabelText("Email") as HTMLInputElement;
    fireEvent.input(emailInputRef, { target: { value: TEST_EMAIL } });
    expect(getAllByDisplayValue(TEST_EMAIL).length).toEqual(1);
  });
  test("Password input allow input and change value", () => {
    const { getByLabelText, getAllByDisplayValue } = component;
    const passwordInputRef = getByLabelText("Password") as HTMLInputElement;
    fireEvent.input(passwordInputRef, { target: { value: TEST_PASSWORD } });
    expect(getAllByDisplayValue(TEST_PASSWORD).length).toEqual(1);
  });

  test("Checkbox allow input to be checked and unchecked", () => {
    const { getByRole } = component;
    const checkboxInputRef = getByRole("checkbox") as HTMLInputElement;
    fireEvent.click(checkboxInputRef);
    expect(checkboxInputRef.checked).toEqual(true);
    fireEvent.click(checkboxInputRef);
    expect(checkboxInputRef.checked).toEqual(false);
  });
});
