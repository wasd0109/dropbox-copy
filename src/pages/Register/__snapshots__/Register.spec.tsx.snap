// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register page render correctly Register page render 1`] = `[Function]`;
